resources/property.py

```diff
@@ -77,7 +77,12 @@ def get(self, id):
         rentalProperty = PropertyModel.find_by_id(id)
 
         if rentalProperty:
-            return rentalProperty.json()
+            json = rentalProperty.json()
+            json["tenants"] = []
+            for lease in rentalProperty.leases:
+                json["tenants"].append(lease.tenant.json())
+            return json
+
```

Comment:
naming things are hard. I think @aedwardg has a good point though. Using the name json doesn't describe what contents the variable is holding. I'm not too concerned about the data structure the variable is, but rather what it is, which is a property object. so I would suggest `property`.

I would suggest the following change to increase readability, and one or two tests. One unit test on the model, and then one integration test on the resource.

If we use the `find` method then we can remove the conditional too.
```
    def get(self, id):
        property = PropertyModel.find(id).json()
        property["tenants"] = property.tenants()
        return property
```

And in the `PropertyModel` add the tenants method.
```
    def tenants(self):
        tenants = []
        for lease in self.leases:
            tenants.append(lease.tenant.json())
        return tenants
```

Reply:
Variable renamed, and code restructured as you suggest!

I'm trying to write a unit test now. I figured you meant unit-testing the new `tenants()` method at the very least -- if that's the case, I ran into an issue, and I'm wondering if my solution is appropriate. The property fixture `create_property()` seems to create a property without leases. To get a property _with_ leases, I actually used the `create_lease()` fixture, and then used the `lease.propertyID` to `find()` the property and check that `tenants()` works the way we expect.

Reply:
Integration test updated, and unit test added. Let me know if this is enough testing -- I'm still getting a feel for exactly what to test! It didn't seem like any of the other property integration tests should be updated, at the very least.